packages/purpose-platformstate-writer/test/consumerServiceV2.integration.test.ts

```diff
@@ -0,0 +1,2540 @@
+import { fail } from "assert";
+import {
+  afterAll,
+  afterEach,
+  beforeAll,
+  beforeEach,
+  describe,
+  expect,
+  it,
+  vi,
+} from "vitest";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import {
+  buildDynamoDBTables,
+  deleteDynamoDBTables,
+  getMockAgreement,
+  getMockDescriptor,
+  getMockPurpose,
+  getMockPurposeVersion,
+  writeTokenStateEntry,
+} from "pagopa-interop-commons-test/index.js";
+import {
+  generateId,
+  itemState,
+  makeGSIPKConsumerIdEServiceId,
+  makeGSIPKEServiceIdDescriptorId,
+  makePlatformStatesAgreementPK,
+  makePlatformStatesEServiceDescriptorPK,
+  makePlatformStatesPurposePK,
+  makeTokenGenerationStatesClientKidPurposePK,
+  NewPurposeVersionActivatedV2,
+  PlatformStatesAgreementEntry,
+  PlatformStatesCatalogEntry,
+  PlatformStatesPurposeEntry,
+  Purpose,
+  PurposeActivatedV2,
+  PurposeArchivedV2,
+  PurposeEventEnvelope,
+  PurposeVersion,
+  PurposeVersionActivatedV2,
+  purposeVersionState,
+  PurposeVersionSuspendedByConsumerV2,
+  PurposeVersionSuspendedByProducerV2,
+  PurposeVersionUnsuspendedByConsumerV2,
+  PurposeVersionUnsuspendedByProducerV2,
+  TokenGenerationStatesClientPurposeEntry,
+  toPurposeV2,
+} from "pagopa-interop-models";
+import { getMockTokenStatesClientPurposeEntry } from "pagopa-interop-commons-test";
+import { handleMessageV2 } from "../src/consumerServiceV2.js";
+import {
+  getPurposeStateFromPurposeVersions,
+  readPlatformPurposeEntry,
+  readAllTokenEntriesByGSIPKPurposeId,
+  writePlatformPurposeEntry,
+} from "../src/utils.js";
+import { config, writeAgreementEntry, writeCatalogEntry } from "./utils.js";
+
+describe("integration tests for events V2", () => {
+  if (!config) {
+    fail();
+  }
+  const dynamoDBClient = new DynamoDBClient({
+    endpoint: `http://localhost:${config.tokenGenerationReadModelDbPort}`,
+  });
+  beforeEach(async () => {
+    await buildDynamoDBTables(dynamoDBClient);
+  });
+  afterEach(async () => {
+    await deleteDynamoDBTables(dynamoDBClient);
+  });
+  const mockDate = new Date();
+  beforeAll(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(mockDate);
+  });
+  afterAll(() => {
+    vi.useRealTimers();
+  });
+
+  describe("PurposeActivated", () => {
```

Comment:
to cover all sue cases, let's add a test for the base case when the record does not exist in platform states (and it is added) and no record exists in token generation states (and nothing is changed)

Reply:
Test added